### DISCUSSION TITLE
ScenegraphLayer Binary Attributes getOrientation?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#5273] [Q&A] [answered]
Looking for some assistance on using the [ScenegraphLayer ](https://deck.gl/docs/api-reference/mesh-layers/scenegraph-layer) with binary attributes. I seem to be having difficulties getting the model's orientation to respond to the array provided by the `getOrientation` function...

I have created a simple [CodeSandbox](https://codesandbox.io/embed/great-euclid-rgs6q) displaying my issue. The aircraft model used in the example simply stays pointed directly towards the ground regardless of values provided from the `getOrientation` function. Looking for a bit of a sanity check as I am wondering if I am perhaps constructing the getOrientation typedArray incorrectly?

I could be mistaken but, looking at the [ScenegraphLayer](https://github.com/visgl/deck.gl/blob/master/modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.js) code the `initializeState` function does not appear to be setting up the `getOrientation` accessor for binary... A bug perhaps?

**Comment by jfuehner** [comment]
@Pessimistress @ibgreen any insights on this question?

**Comment by Pessimistress** [comment, marked as answer]
`getOrientation` is not [directly filled into an attribute](https://github.com/visgl/deck.gl/blob/master/modules/mesh-layers/src/utils/matrix.js#L8), so in that sense, it doesn't support binary attributes.

**Reply by jfuehner** [reply]
@Pessimistress I see... Is that a bug or would you recommend I file a feature request for the ScenegraphLayer to support orientation as a binary attribute? 

Seems like if some of the attributes (i.e., color, position, etc) support binary then the others (i.e., orientation, scale, transformation, etc) would support binary as well...

**Reply by Pessimistress** [reply]
It's not a bug, just a limitation. The layer's documentation should make this more clear for sure.

Even if this accessor supports binary input, it still needs to do this processing on the CPU side, so it defeats the purpose if you are aiming for better performance.

**Reply by jfuehner** [reply]
@Pessimistress thanks for the explanation; it’s helpful to know it is a limitation and the processing is CPU bound. 

Was hoping to avoid the CPU as I have done with other layers.
